torch/distributed/_sharded_tensor/ops/elementwise_ops.py

```diff
@@ -0,0 +1,35 @@
+import copy
+
+import torch
+from torch.distributed._sharded_tensor import (
+    sharded_op_impl,
+    Shard,
+    ShardedTensor,
+)
+
+
+@sharded_op_impl(torch.nn.functional.gelu)
+def gelu(types, args=(), kwargs=None, pg=None):
+    """
+    Handles ``__torch_function__`` dispatch for the elementwise op
+    ``torch.nn.functional.gelu``.
+    This method computes on either a normal tensor or a sharded tensor.
+    """
+    input = args[0]
+    op = torch.nn.functional.gelu
+    # Validate types
+    if not isinstance(input, torch.Tensor) and not isinstance(input, ShardedTensor):
+        raise TypeError("input needs to be either torch.Tensor or ShardedTensor")
+    if isinstance(input, torch.Tensor):
+        return op(args[0])
+    else:
+        local_shards_new = []
+        for local_shard in input.local_shards():
+            local_shards_new.append(Shard(op(local_shard.tensor), local_shard.metadata))
+        new_st = ShardedTensor._init_from_local_shards(
+            local_shards_new, input.size(), process_group=pg
+        )
```

Comment:
This method actually performs an allgather: https://github.com/pytorch/pytorch/blob/master/torch/distributed/_sharded_tensor/api.py#L351. We probably need a variation of this that doesn't do the validation for perf reasons or have a separate simpler API.

For now, this is fine but we should improve upon this in follow up PRs once we have an API to do this efficiently.

cc @wanchaol

Reply:
As we discussed offline, will create a new API named `_init_from_local_tensor` to avoid the potential cost of `all_gather_object`.